modules/kvtagger/tests/test_kvtagdb.c

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright (c) 2016 Balabit
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "kvtagdb.h"
+#include "testutils.h"
+#include <stdio.h>
+#include <string.h>
+
+#define KVTAGDB_TESTCASE(testfunc, ...) {testcase_begin("%s(%s)", #testfunc, #__VA_ARGS__); testfunc(__VA_ARGS__); testcase_end();}
+
+static void
+_foreach_ctr(gpointer arg, const TagRecord *record)
+{
+  int *ctr = (int *)arg;
+  ++(*ctr);
+}
+
+static void
+_test_empty_db(KVTagDB *tagdb)
+{
+  assert_false(kvtagdb_is_loaded(tagdb) == TRUE, "Empty KVTagDB should be in unloaded state.");
+  assert_false(kvtagdb_is_indexed(tagdb) == TRUE, "Empty KVTagDB should be in un-indexed state.");
+  assert_false(kvtagdb_contains(tagdb, "selector") == TRUE, "Method kvtagdb_contains should work with empty KVTagDB.");
+  assert_true(kvtagdb_number_of_tags(tagdb, "selector") == 0, "Method kvtagdb_number should work with empty KVTagDB.");
+  int ctr = 0;
+  kvtagdb_foreach_tag(tagdb, "selector", _foreach_ctr, (gpointer)&ctr);
+  assert_true(ctr == 0, "Method kvtagdb_foreach_tag should work for with empty KVTagDB.");
+}
+
+static void
+test_empty_db()
```

Comment:
The difference between `_test_empty_db(void)` and `test_empty_db(void)` is that one calls the other. Perhaps some more descriptive name? (and missing underscore as noted earlier)